src/Security/MemberAuthenticator/LostPasswordHandler.php

```diff
@@ -195,15 +187,45 @@ public function forgotPassword($data, $form)
             $this->sendEmail($member, $token);
         }
 
-        // Avoid information disclosure by displaying the same status,
-        // regardless wether the email address actually exists
-        $link = Controller::join_links(
-            $this->link('passwordsent'),
-            rawurlencode($data['Email']),
-            '/'
-        );
+        return $this->redirectToSuccess($data);
+    }
 
-        return $this->redirect($this->addBackURLParam($link));
+    /**
+     * Ensure that the user has provided an email address. Note that the "Email" key is specific to this
+     * implementation, but child classes can override this method to use another unique identifier field
+     * for validation.
+     *
+     * @param  array $data
+     * @param  LostPasswordForm $form
+     * @return HTTPResponse|null
+     */
+    protected function validateForgotPasswordData(array $data, LostPasswordForm $form)
+    {
+        if (empty($data['Email'])) {
+            $form->sessionMessage(
+                _t(
+                    'SilverStripe\\Security\\Member.ENTEREMAIL',
+                    'Please enter an email address to get a password reset link.'
+                ),
+                'bad'
+            );
+
+            return $this->redirectToLostPassword();
+        }
+    }
+
+    /**
+     * Load an existing Member from the provided data
+     *
+     * @param  array $data
+     * @param  string $uniqueIdentifier
+     * @return Member|null
+     */
+    protected function getMemberFromData(array $data, $uniqueIdentifier)
```

Comment:
I just realised this second arg isn't even needed. Just pull it in directly from config.

```
$uniqueIdentifier = Member::config()->get('unique_identifier_field');
```

Reply:
Cool, updated and rebased